packages/edit-navigation/src/index.js

```diff
@@ -16,13 +16,21 @@ import { __ } from '@wordpress/i18n';
 import apiFetch from '@wordpress/api-fetch';
 import { addQueryArgs } from '@wordpress/url';
 import { decodeEntities } from '@wordpress/html-entities';
+import { addFilter } from '@wordpress/hooks';
 
 /**
  * Internal dependencies
  */
 import Layout from './components/layout';
 import './store';
 
+function disableInsertingNonNavigationBlocks( settings, name ) {
+	if ( ! [ 'core/navigation', 'core/navigation-link' ].includes( name ) ) {
+		set( settings, [ 'supports', 'inserter' ], false );
+	}
+	return settings;
+}
+
```

Comment:
Could also just register the blocks we want instead of registering every core and experimental block and then disabling inserter support.

Reply:
The idea is to have them registered but not insertable so that, when a user switches away from a theme that supports blocks, they'll still see the blocks that they previously added to their menu. This way we're not "losing" any of the user's data. 

We'll need to add some sort of UI which indicates that the invalid blocks won't display in the frontend. I think best to do that separately since it requires design.

Reply:
I see what you mean, though not registering the blocks seems fairly close to what you describe. If they're not registered, we wouldn't lose the data, but the block would be shown as unrecognized/missing.

For this page we could also consider registering a different 'missing' block that handles the messaging better:
https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src/missing

Reply:
> For this page we could also consider registering a different 'missing' block that handles the messaging better:
https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src/missing

Good idea!

> Perhaps move this particular part out into a separate issue.

Agreed.